cli/cmd/compliance.go

```diff
@@ -263,3 +283,64 @@ func buildComplianceReportTable(detailsTable, summaryTable, recommendationsTable
 	}
 	return mainReport.String()
 }
+
+func filterRecommendations(recommendations []api.ComplianceRecommendation) ([]api.ComplianceRecommendation, string) {
+	var filtered []api.ComplianceRecommendation
+	for _, r := range recommendations {
+		if matchRecommendationsFilters(r) {
+			filtered = append(filtered, r)
+		}
+	}
+	if len(filtered) == 0 {
+		return filtered, "There are no recommendations with the specified filters.\n"
+	}
+
+	cli.Log.Debugw("filtered recommendations", "recommendations", filtered)
+	return filtered, fmt.Sprintf("%v of %v recommendations showing \n", len(filtered), len(recommendations))
+}
+
+func matchRecommendationsFilters(r api.ComplianceRecommendation) bool {
+	var results []bool
+
+	// severity returns specified threshold and above
+	if compCmdState.Severity != "" {
+		sevThreshold, _ := eventSeverityToProperTypes(compCmdState.Severity)
+		results = append(results, r.Severity <= sevThreshold)
+	}
+
+	if compCmdState.Category != "" {
+		category := strings.ReplaceAll(compCmdState.Category, "-", " ")
+		results = append(results, strings.EqualFold(r.Category, category))
+	}
+
+	if compCmdState.Service != "" {
+		results = append(results, strings.EqualFold(r.Service, compCmdState.Service))
+	}
+
+	if compCmdState.Status != "" {
+		results = append(results, r.Status == statusToProperTypes(compCmdState.Status))
+	}
+
+	return !array.ContainsBool(results, false)
+}
+
+func filtersEnabled() bool {
+	return compCmdState.Category != "" || compCmdState.Status != "" || compCmdState.Severity != "" || compCmdState.Service != ""
+}
+
+func statusToProperTypes(status string) string {
+	switch strings.ToLower(status) {
+	case "non-compliant", "noncompliant":
+		return "NonCompliant"
+	case "compliant":
+		return "Compliant"
+	case "could-not-assess":
```

Comment:
just because you have the other two, maybe have them all be consistent.
```suggestion
	case "could-not-assess", "couldnotassess":
```